programs/github

```diff
@@ -21,13 +21,61 @@ function progressbar(f)
 	return left .. right
 end
 
+-- get flag arguments
+local function findFlag(arg)
+	local accepted = {
+		{ name = 'branch', flags = { '-b', '--branch' } },
+		{ name = 'tag', flags = { '-t', '--tag' } },
+		{ name = 'auth', flags = { '-a', '--auth' } }
+	}
+	local isFlag, flagName = false, nil
+	if arg:sub(1,1) == '-' then
+		for _, flagType in pairs(accepted) do
+			for _, flag in pairs(flagType.flags) do
+				if arg == flag then
+					isFlag = true
+					flagName = flagType.name
+					return isFlag, flagName
+				end
+			end
+		end
+		if not isFlag then
```

Comment:
right **if** it runs the whole intent is we have an argument that starts with '-' but wasn't found in the list of flags we accept.